packages/eslint-plugin/src/lib/rules/require-http-module-factory/index.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * @fileoverview Rule to require HttpModule to use registerAsync factory to avoid shared axios client issues
```

Comment:
Devinai add a README.md file with a more verbose description about the rule and motivation.